storage_service/locations/models/rclone.py

```
@@ -0,0 +1,239 @@
+                    LOGGER.warning(stderr)
+
+                return stdout, stderr
+        except FileNotFoundError as err:
+            err_msg = "rclone executable not found at path. Details: {}".format(err)
+            LOGGER.error(err_msg)
+            raise StorageException(err_msg)
+        except Exception as err:
+            err_msg = (
+                "Error running rclone command. Command called: {}. Details: {}".format(
+                    cmd, err
+                )
+            )
+            LOGGER.error(err_msg)
+            raise StorageException(err_msg)
+
+    @property
+    def remote_prefix(self):
+        """Return remote prefix from env vars matching case-insensitive RClone.remote_name."""
+        remotes, _ = self._execute_subprocess(["listremotes"])
+        LOGGER.debug("rclone listremotes output: %s", remotes)
+        remotes = six.ensure_str(remotes).split("\n")
+
+        for remote in remotes:
+            if remote.startswith(self.remote_name.lower()):
+                LOGGER.debug("rclone remote selected: %s", remote)
+                return remote
+
+        raise StorageException("rclone remote matching %s not found", self.remote_name)
+
+    def _ensure_container_exists(self):
+        """Ensure that the S3 bucket or other container exists by asking it
+        something about itself. If we cannot retrieve metadata about it then
+        we attempt to create the bucket, else, we raise a StorageException.
+        """
+        LOGGER.debug("Test that container '%s' exists", self.container)
+        prefixed_container_name = "{}{}".format(self.remote_prefix, self.container)
+        cmd = ["ls", prefixed_container_name]
+        _, stderr = self._execute_subprocess(cmd)
+        if stderr:
+            LOGGER.info("Creating container '%s'", self.container)
+            create_container_cmd = ["mkdir", prefixed_container_name]
+            _, stderr = self._execute_subprocess(create_container_cmd)
+            if stderr:
```

Should we raise the exception based on the return code of rclone, instead of stderr?

When the config file doesn't exist , seems like a NOTICE is logged trough stderr, thus breaking the process.

---

Yeah that makes sense to me! I'll make that change today.

---

Change made in commit https://github.com/artefactual/archivematica-storage-service/pull/635/commits/5a17501b25d10845c6e0f655b706756d0713198c